src/utils/os.c

```diff
@@ -286,6 +286,40 @@ int read_command_output(int fd, process_callback_fn fn, void *ctx) {
   return count;
 }
 
+char **copy_argv(const char *const argv[]) {
+  // calculate the length of argv (without NULL terminator)
```

Comment:
```c
if (argv == NULL) {
  log_error("argv param is NULL");
  return NULL;
}
```

Reply:
Thanks for the suggestion.

By the way, you can use the following syntax so that GitHub has the fancy "Commit suggestion":

````markdown
```suggestion
you're suggestion here
```
````

Reply:
Thanks for reminding me. I always forget about this trick.